Test/monadic/MonadicBind.dfy

```diff
@@ -0,0 +1,77 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
```

Comment:
I would like to see more tests: some that don't use specifications, some that use `reads`.
Also, since these are expected to verify, there might as well be some compiling tests (so change to `/compile:3` and add a `Main`).